src/main/java/de/tum/in/www1/artemis/service/iris/session/IrisCodeEditorSessionService.java

```diff
@@ -118,32 +118,19 @@ public IrisCodeEditorSession createSession(ProgrammingExercise exercise, User us
      * Checks if the user has access to the Iris session. A user has access if they have access to the exercise and the
      * session belongs to them. If the user is null, the user is fetched from the database.
      *
-     * @param session The session to check
      * @param user    The user to check
+     * @param session The session to check
      */
     @Override
-    public void checkHasAccessToIrisSession(IrisSession session, User user) {
-        checkHasAccessToIrisSession(castToSessionType(session, IrisCodeEditorSession.class), user);
-    }
-
-    private void checkHasAccessToIrisSession(IrisCodeEditorSession session, User user) {
+    public void checkHasAccessTo(User user, IrisCodeEditorSession session) {
         authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, session.getExercise(), user);
         if (!Objects.equals(session.getUser(), user)) {
             throw new AccessForbiddenException("Iris Code Editor Session", session.getId());
         }
     }
 
     @Override
-    public void checkRateLimit(User user) {
-        // Currently no rate limit for code editor sessions
-    }
-
-    @Override
-    public void checkIsIrisActivated(IrisSession session) {
-        checkIsIrisActivated(castToSessionType(session, IrisCodeEditorSession.class));
-    }
-
-    private void checkIsIrisActivated(IrisCodeEditorSession session) {
+    public void checkIsFeatureActivatedFor(IrisCodeEditorSession session) {
```

Comment:
Could you add JavaDoc since this is a public method?

Reply:
Done :)